packages/flutter_tools/lib/src/flx.dart

```diff
@@ -200,6 +200,34 @@ Map<_Asset, List<_Asset>> _parseAssets(
   return result;
 }
 
+final String _licenseSeparator = '\n' + ('-' * 80) + '\n';
+
+/// Returns a ZipEntry representing the license file.
+ZipEntry _obtainLicenses(
+  PackageMap packageMap,
+  String assetBase
+) {
+  // Read the LICENSE file from each package in the .packages file,
+  // splitting each one into each component license (so that we can
+  // de-dupe if possible).
+  final Set<String> packageLicenses = new Set<String>();
+  final List<Uri> packages = packageMap.map.values;
+  for (Uri package in packages) {
+    if (package != null && package.scheme == 'file') {
+      final File file = new File.fromUri(package.resolve('../LICENSE'));
+      if (file.existsSync())
+        packageLicenses.addAll(file.readAsStringSync().split(_licenseSeparator));
+    }
+  }
+
+  final List<String> combinedLicensesList = packageLicenses.toList();
+  combinedLicensesList.sort();
```

Comment:
Should we de-dupe here? Get some normalized fingerprint of the license text and remove duplicates based on that fingerprint?